Task 8/UsersAndAwards(Framework)/EPAM.AwardsAndUsers.BLL.JSONBLL/JsonLogic.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using EPAM.AwardsAndUsers.BLL.Interfaces;
+using EPAM.AwardsAndUsers.Common.Entities;
+using EPAM.AwardsAndUsers.DAL.Interfaces;
+
+namespace EPAM.AwardsAndUsers.BLL.JSONBLL
+{
+    public class JsonLogic: ILogicBLL
+    {
+        private IDAL _daoLogic;
+
+        public JsonLogic(IDAL daoLogic)
+        {
+            _daoLogic = daoLogic;
+        }
+
+        public void SetBase()
+        {
+            _daoLogic.SetBase();
+        }
+
+        public void AddUser(User user)
+        {
+            Data data = _daoLogic.LoadData();
+            data.AddKey(user.id);
+            _daoLogic.RecordUserToFile(user);
+            _daoLogic.RecordData(data);
+        }
+
+        public void AddAward(Award award)
+        {
+            _daoLogic.RecordAwardToFile(award);
+        }
+
+        public void RemoveUser(Guid id)
+        {
+            Data data = _daoLogic.LoadData();
+            User user = _daoLogic.GetAllUsers().First(item => item.id == id);
+            if (data.DataValue.Remove(id))
+                _daoLogic.RecordData(data);
+            _daoLogic.RemoveUser(id);
+            _daoLogic.RemoveAuthData(id);
+            _daoLogic.RemoveRolesData(user.Name);
```

Comment:
It contains a lot of data logic, data logic should be on DAL level. It's okay that your current BLL has near to no logic - this project doesn't require much business logic. 
BUT
You can implement validation here (like if user doesn't exists or if Award has empty name, etc